src/TuDa.CIMS.Api/Controllers/PurchaseController.cs

```diff
@@ -103,4 +103,12 @@ [FromBody] CreatePurchaseDto createModel
             await _purchaseService.InvalidateAsync(workingGroupId, purchaseId, createModel)
         ).Match(onValue: _ => Ok(), onError: ErrorsToProblem);
     }
+
+    [HttpGet($"{{{nameof(purchaseId)}:guid}}/signature")]
+    public async Task<IActionResult> RetrieveSignatureAsync(Guid workingGroupId, Guid purchaseId)
+    {
+        return (
+            await _purchaseService.RetrieveSignatureAsync(workingGroupId, purchaseId)
+        ).Match(onValue: _ => Ok(), onError: ErrorsToProblem);
```

Comment:
🔧 You are not returning the signature here

Reply:
We should use something like this 
```suggestion
        ).Match(onValue: value => File(value, "image/png");, onError: ErrorsToProblem);
```

This returns a concrete `png` image on the endpoint